src/d/actor/d_a_pw.cpp

```diff
@@ -107,8 +111,253 @@ void first_mode_change(pw_class*) {
 }
 
 /* 00002A90-00003B08       .text action_dousa__FP8pw_class */
-void action_dousa(pw_class*) {
+void action_dousa(pw_class* i_this) {
     /* Nonmatching */
+    daPy_py_c* link = daPy_getPlayerActorClass();
+    float randomValue;
+    switch (i_this->m368) {
+    case 0:
+        anm_init(i_this, 0x23, 7.0, '\x02', 1.0, -1);
+        i_this->m346 = 0;
+        i_this->m368 += 1;
+        break;
+
+    case 0x1:
+        f32 actor_distance = fopAcM_searchActorDistance(i_this, link);
+        if (actor_distance < 500.0f) {
+            anm_init(i_this, 0x10, 3.0, '\0', 1.0, -1);
+            s16 actorYAngle = fopAcM_searchActorAngleY(i_this, link);
```

Comment:
```suggestion
            i_this->m38C = fopAcM_searchActorAngleY(i_this, link);
```
I don't think there's any reason to put this in a temporary local variable before assigning it, unless that's required for this to match?